mne/viz/_3d.py

```diff
@@ -957,6 +1046,185 @@ def time_label(t):
+        for possible values. Defaults to '2darrow'.
+    scale_factor : float | None
+        Dipoles are drawn at a size proportional to the magnitude of the
+        current at the corresponding source vertex. This parameter controls the
+        overall size of the dipoles in the plot. If None, an automatic scaling
+        will be attempted. Defaults to None.
+    hemi : 'lh' | 'rh' | 'both'
+        The hemisphere to plot. Defaults to 'both'.
+    colorbar : bool
+        If True, display colorbar on scene. Defaults to True.
+    cmap : str
+        The Mayavi colormap to use. Defaults to 'mne'.
+    clim : str | dict
+        Colorbar properties specification. If 'auto', set clim automatically
+        based on data percentiles. If dict, should contain:
+            ``kind`` : str
+                Flag to specify type of limits. 'value' or 'percent'.
+            ``lims`` : list | np.ndarray | tuple of float, 3 elements
+                Note: Only use this if 'colormap' is not 'mne'.
+                Left, middle, and right bound for colormap.
+            ``pos_lims`` : list | np.ndarray | tuple of float, 3 elements
+                Note: Only use this if 'colormap' is 'mne'.
+                Left, middle, and right bound for colormap. Positive values
+                will be mirrored directly across zero during colormap
+                construction to obtain negative control points.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    fig : instance of mlab.Figure
+        The mayavi figure.
+    """
+    from mayavi import mlab
+
+    fig = mlab.figure(size=fig_size, bgcolor=bgcolor, fgcolor=(0, 0, 0))
+
+    t = np.searchsorted(stc.times, time)
+    magnitude = stc.magnitude()
+
+    scale_pts, colormap, transparent = _limits_to_control_points(
+        clim, stc.data, cmap, transparent)
+
+    # The colomap can be specified both as a LUT table and a string
+    if isinstance(colormap, np.ndarray):
+        lut_table = colormap
+        cmap = 'hot'
+    else:
+        lut_table = None
+        cmap = colormap
+
+    if scale_factor is None:
+        scale_factor = 1. / (80 * np.max(magnitude.data[:, t]))
+
+    def _plot_hemi_mesh(src, hemi_ind):
+        """Plot the mesh of a single hemisphere"""
+        with warnings.catch_warnings(record=True):  # FutureWarning in traits
+            if high_resolution:
+                tris = src[hemi_ind]['tris']
+            else:
+                tris = src[hemi_ind]['use_tris']
+
+            # Plot mesh
+            mlab.triangular_mesh(src[hemi_ind]['rr'][:, 0],
+                                 src[hemi_ind]['rr'][:, 1],
+                                 src[hemi_ind]['rr'][:, 2],
+                                 tris,
+                                 color=brain_color,
+                                 opacity=opacity,
+                                 representation='surface')
```

Comment:
Looking more into using PySurfer, I'm now less enthusiastic about it. PySurfer will return a `Brain` object which allows the user to control the plot (changing the colormap, scrolling through time, etc). The plotting of a `VectorSourceEstimate` will support none of that and I don't know how to handle this elegantly. Improving the plotting function can maybe be addressed in a later PR?